test/bench/synthetic_benchmarks.cpp

```diff
@@ -0,0 +1,158 @@
+
+#include "synthetic_benchmarks.hpp"
+#include "helpers.hpp"
+#include "test/utils/bytecode.hpp"
+#include <evmc/instructions.h>
+#include <evmone/instruction_traits.hpp>
+
+using namespace benchmark;
+
+namespace evmone::test
+{
+namespace
+{
+/// Stack limit inside the EVM benchmarking loop (one stack item is used for the loop count).
+constexpr auto stack_limit = 1023;
+
+enum class Mode
+{
+    min_stack = 0,   ///< The code uses as minimal stack as possible.
+    full_stack = 1,  ///< The code fill the stuck up to its limit.
+};
+
+enum class InstructionKind : char
+{
+    unop = 'u',
+    binop = 'b',
+    push = 'p',
+    producer = 'a',
+    unknown = 'X',
+};
+
+constexpr InstructionKind get_instruction_kind(evmc_opcode opcode) noexcept
+{
+    const auto trait = instr::traits[opcode];
+    if (opcode >= OP_PUSH1 && opcode <= OP_PUSH32)
+        return InstructionKind::push;
+    else if (trait.stack_height_required == 1 && trait.stack_height_change == 0)
+        return InstructionKind::unop;
+    else if (trait.stack_height_required == 2 && trait.stack_height_change == -1)
+        return InstructionKind::binop;
+    else if (trait.stack_height_required == 0 && trait.stack_height_change == 1)
+        return InstructionKind::producer;
+    else
+        return InstructionKind::unknown;
+}
+
+/// Generates the EVM benchmarking loop inner code for the given opcode and "mode".
+bytecode generate_loop_inner_code(evmc_opcode opcode, Mode mode)
+{
+    const auto kind = get_instruction_kind(opcode);
+    switch (mode)
+    {
+    case Mode::min_stack:
+        switch (kind)
+        {
+        case InstructionKind::push:
+            return stack_limit * (push(opcode, {}) + OP_POP);
+        case InstructionKind::producer:
+            return stack_limit * (bytecode{opcode} + OP_POP);
+        case InstructionKind::unop:
+            return OP_DUP1 + stack_limit * 2 * bytecode{opcode} + OP_POP;
+        case InstructionKind::binop:
+            return OP_DUP1 + (stack_limit - 1) * (OP_DUP1 + bytecode{opcode}) + OP_POP;
+        default:
+            INTX_UNREACHABLE();
+        }
+    case Mode::full_stack:
+        switch (kind)
+        {
+        case InstructionKind::push:
+            return stack_limit * push(opcode, {}) + stack_limit * OP_POP;
+        case InstructionKind::producer:
+            return stack_limit * opcode + stack_limit * OP_POP;
+        case InstructionKind::binop:
+            return stack_limit * OP_DUP1 + (stack_limit - 1) * opcode + OP_POP;
+        default:
+            INTX_UNREACHABLE();
+        }
+    default:
+        INTX_UNREACHABLE();
+    }
+
+    return {};  // Make old compilers happy.
+}
+
+
+const auto loop_prefix = push(255) + OP_JUMPDEST;
+const auto loop_suffix = push("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff") +
+                         OP_ADD + OP_DUP1 + push(2) + OP_JUMPI;
+
+bytes_view generate_code(evmc_opcode opcode, Mode mode) noexcept
```

Comment:
`generate_loop_inner_code` is not `noexcept`
```suggestion
bytes_view generate_code(evmc_opcode opcode, Mode mode)
```